### ISSUE TITLE
Bbolt's read-only status is interpreted as a disk error

### ISSUE THREAD

**Issue by carpawell** [nspcc-dev/neofs-node#2057]
Some race is here. Also, I doubt moving DB to read-only mode by *a node* is a disk error.

```
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.149Z        debug        blobovniczatree/control.go:24        read-only mode, skip blobovniczas initialization...
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.149Z        info        engine/engine.go:105        shard is moved in degraded mode due to error threshold        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 30}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.158Z        warn        engine/engine.go:153        could not put object to active blobovnicza        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 31, "error": "database not open"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.158Z        debug        blobovnicza/control.go:17        creating directory for BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/3/0", "ro": true}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.158Z        debug        blobovnicza/control.go:31        opening BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/3/0", "permissions": "-rwxrwxrwx"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.158Z        debug        blobovniczatree/blobovnicza.go:201        blobovnicza successfully activated        {"path": "3/0"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.158Z        info        engine/engine.go:105        shard is moved in degraded mode due to error threshold        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 31}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.169Z        warn        engine/engine.go:153        could not put object to active blobovnicza        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 32, "error": "database is in read-only mode"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.169Z        debug        blobovnicza/control.go:17        creating directory for BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/1/0", "ro": true}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.169Z        debug        blobovnicza/control.go:31        opening BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/1/0", "permissions": "-rwxrwxrwx"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.169Z        debug        blobovniczatree/blobovnicza.go:201        blobovnicza successfully activated        {"path": "1/0"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.169Z        info        engine/engine.go:105        shard is moved in degraded mode due to error threshold        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 32}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.179Z        warn        engine/engine.go:153        could not put object to active blobovnicza        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 33, "error": "database is in read-only mode"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.179Z        debug        blobovnicza/control.go:17        creating directory for BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/2/0", "ro": true}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.179Z        debug        blobovnicza/control.go:31        opening BoltDB        {"component": "Blobovnicza", "path": "/srv/neofs/data0/blobovnicza/2/0", "permissions": "-rwxrwxrwx"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.179Z        debug        blobovniczatree/blobovnicza.go:201        blobovnicza successfully activated        {"path": "2/0"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.179Z        info        engine/engine.go:105        shard is moved in degraded mode due to error threshold        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 33}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.190Z        warn        engine/engine.go:153        could not put object to active blobovnicza        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 34, "error": "database is in read-only mode"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.190Z        warn        engine/engine.go:153        can't flush an object to blobstor        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 35, "error": "could not save the object in any blobovnicza"}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.190Z        debug        engine/engine.go:176        mode change is in progress, ignoring set-mode request        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error_count": 35}
Nov 14 02:42:42 az neofs-node[14468]: 2022-11-14T02:42:42.190Z        info        engine/engine.go:105        shard is moved in degraded mode due to error threshold        {"shard_id": "UdNgukPGsHGaTvDp7tGtz3", "error count": 34}
```

**Comment by fyrchik** [comment]
We could take mode mutex on read for all operations and return pre-defined `ErrReadOnly` error.

**Comment by fyrchik** [comment]
I coudn't come up with an example of then this can happen (after the fix from my previous comment).

**Comment by carpawell** [comment]
Closed via #2059.
